src/coreclr/inc/corinfo.h

```diff
@@ -1982,6 +1982,14 @@ enum class GetTypeLayoutResult
     Failure,
 };
 
+struct CORINFO_SWIFT_LOWERING
+{
+    bool byReference;
+    CorInfoType loweredElements[4];
+    uint32_t offsets[4];
+    size_t numLoweredElements;
+};
```

Comment:
Is there value in documenting this type or fields? The `byReference` seems clear - I assume `in`|`ref`|`out`. Why only 4 fields, is that something we can/should capture in a comment related to the Swift ABI?

I would prefer to see the `numLoweredElements` to be before the arrays. I like reading fields in order and I assume callers will read the count before the array.

Reply:
Okay, let's keep the layout style consistent then.

Reply:
> Is there value in documenting this type or fields?

We could reference the Swift Interop design doc and the Swift ABI doc, but I'd be more inclined to document the algorithm implementation with this information than the struct since the struct is just whatever shape the JIT team likes the best.

> The `byReference` seems clear - I assume `in`|`ref`|`out`. Why only 4 fields, is that something we can/should capture in a comment related to the Swift ABI?

`byReference` actually means that instead of being lowered to primitives, the struct will be passed as a reference to its location on the stack. It's not `in/ref/out` in this case.